litho-widget/src/main/java/com/facebook/litho/widget/RenderInfoViewCreatorController.java

```diff
@@ -25,31 +25,55 @@
   @VisibleForTesting
   final SimpleArrayMap<ViewCreator, Integer> mViewCreatorToViewType = new SimpleArrayMap<>();
 
+  public static final int DEFAULT_COMPONENT_VIEW_TYPE = 0;
+  private final boolean mCustomViewTypes;
+  private final int mComponentViewType;
+  private int mViewTypeCounter;
 
-  public static final int COMPONENT_VIEW_TYPE = 0;
-  private int mViewTypeCounter = COMPONENT_VIEW_TYPE + 1;
+  public RenderInfoViewCreatorController(boolean hasCustomViewType, int componentViewType) {
+    mCustomViewTypes = hasCustomViewType;
+    mComponentViewType = componentViewType;
+    mViewTypeCounter = componentViewType + 1;
+  }
 
   @UiThread
   public void maybeTrackViewCreator(RenderInfo renderInfo) {
     if (!renderInfo.rendersView()) {
       return;
     }
+    if (mCustomViewTypes && !renderInfo.hasCustomViewType()) {
+      throw new IllegalStateException("If you enable custom viewTypes, you must provide a"
+          + " customViewType in ViewRenderInfo.");
+    } else if (!mCustomViewTypes && renderInfo.hasCustomViewType()) {
+      throw new IllegalStateException("You must enable custom viewTypes to provide customViewType"
+          + " in ViewRenderInfo.");
+    }
```

Comment:
Can you write tests that cover these (four) combinations: Custom viewtype enabled/disabled + user provides/doesn't provide viewtype? Like you could create `RecyclerBinder` insert some items and expect exception.